datasets/flwr_datasets/partitioner/continuous_partitioner_test.py

```diff
@@ -0,0 +1,188 @@
+
+
+def _dummy_setup(
+    num_partitions: int,
+    strictness: float,
+    num_rows: int,
+    partition_by: str = "score",
+    shuffle: bool = True,
+    seed: Optional[int] = 42,
+) -> tuple[Dataset, ContinuousPartitioner]:
+    """Create a dummy dataset and partitioner for testing."""
+    scores = np.linspace(0, 10, num_rows)
+    data = {
+        partition_by: scores.tolist(),
+        "features": np.random.randn(num_rows).tolist(),
+    }
+    dataset = Dataset.from_dict(data)
+    partitioner = ContinuousPartitioner(
+        num_partitions=num_partitions,
+        partition_by=partition_by,
+        strictness=strictness,
+        shuffle=shuffle,
+        seed=seed,
+    )
+    partitioner.dataset = dataset
+    return dataset, partitioner
+
+
+class TestContinuousPartitionerSuccess(unittest.TestCase):
+    """Test ContinuousPartitioner used with no exceptions."""
+
+    @parameterized.expand(
+        [
+            (5, 0.0, 100),
+            (5, 1.0, 100),
+            (3, 0.5, 50),
+        ]
+    )  # type: ignore
+    def test_valid_partition_shapes(
+        self, num_partitions: int, strictness: float, num_rows: int
+    ) -> None:
+        """Check each partition has non-zero and collectively complete indices."""
+        _, partitioner = _dummy_setup(num_partitions, strictness, num_rows)
+        partition_sizes = [
+            len(partitioner.load_partition(pid)) for pid in range(num_partitions)
+        ]
+        self.assertEqual(sum(partition_sizes), num_rows)
+        self.assertTrue(all(size > 0 for size in partition_sizes))
+
+    def test_partition_determinism_with_fixed_seed(self) -> None:
+        """Check partitioning is deterministic given fixed seed and shuffle=True."""
+        _, partitioner_1 = _dummy_setup(4, 0.7, 40, seed=123)
+        _, partitioner_2 = _dummy_setup(4, 0.7, 40, seed=123)
+        idx1 = partitioner_1.partition_id_to_indices
+        idx2 = partitioner_2.partition_id_to_indices
+        self.assertEqual(idx1, idx2)
+
+    def test_partitioning_different_strictness(self) -> None:
+        """Check different strictness lead to different groupings."""
+        _, p_lo = _dummy_setup(4, 0.1, 40, seed=42)
+        _, p_hi = _dummy_setup(4, 0.9, 40, seed=42)
+        self.assertNotEqual(p_lo.partition_id_to_indices, p_hi.partition_id_to_indices)
+
+    def test_num_partitions_property(self) -> None:
+        """Test num_partitions property returns correct value."""
+        _, partitioner = _dummy_setup(3, 0.5, 30)
+        self.assertEqual(partitioner.num_partitions, 3)
+
+    def test_shuffle_false_gives_sorted_groups(self) -> None:
+        """Test when shuffle=False, partition indices are sorted."""
+        _, partitioner = _dummy_setup(3, 1.0, 30, shuffle=False)
+        for indices in partitioner.partition_id_to_indices.values():
+            self.assertTrue(indices == sorted(indices))
+
+    def test_strictness_zero_is_pure_noise(self) -> None:
```

Comment:
I'm not convinced by how this is being tested. Wouldn't it be better to measure if these partitions are equivalent to a uniform sampling of the whole dataset ? Something along those lines? wdyt?